src/toil/jobStores/abstractJobStore.py

```diff
@@ -114,6 +158,81 @@ def _checkJobStoreCreation(create, exists, jobStoreString):
             raise JobStoreCreationException("The job store '%s' does not exist, so there "
                                             "is nothing to restart." % jobStoreString)
 
+    def importFile(self, srcUrl):
+        """
+        Imports the file pointed at by sourceUrl into jobStore. The jobStoreFileId of the new
+        file is returned.
+
+        :param str sourceUrl: URL that points to either a local file or to a file in a supported
+                jobStore e.g. 's3' for AWS.
+        :return The jobStoreFileId of imported file.
+        """
+        with self.writeFileStream() as (writable, jobStoreFileID):
+            findJobStoreForUrl(srcUrl).writeFromUrl(srcUrl, writable)
+            return jobStoreFileID
+
+    def exportFile(self, jobStoreFileID, dstUrl):
+        """
+        Exports file to destination pointed at by the destUrl.
+
+        :param str jobStoreFileID: The id of the file in the jobStore that should be exported.
+        :param str destUrl: URL that points to either a local file or to a file in a supported
+                jobStore e.g. 's3' for AWS.
+        """
+        with self.readFileStream(jobStoreFileID) as readable:
+            findJobStoreForUrl(dstUrl).writeToUrl(readable, dstUrl)
+
+    @abstractstaticmethod
+    def writeFromUrl(url, writable):
+        """
+        This method takes a url and a writable stream and writes the contents of the file
+        pointed at by the url to that stream.
+
+        :param str url: url pointing to a file in a jobStore
+        :param writable: a writable stream
+        """
+        raise NotImplementedError()
+
+    @abstractstaticmethod
+    def writeToUrl(readable, url):
+        """
+        This method takes a url and a readable stream and writes the contents from the stream
+        to the file pointed at by the url.
+
+        :param str url: url pointing to a file in a jobStore
+        :param readable: a readable stream
+        """
+        raise NotImplementedError()
+
+    @classmethod
+    def supportsUrl(cls, url):
+        """
+        This method returns True if a JobStore supports the scheme of the given url.
+
+        :param str url: a url string that may be supported
+        :return bool: returns true if the cls supports the url
+        """
+        return cls.nativeUrlRegex().match(url) is not None
+
+    @abstractclassmethod
+    def nativeUrlRegex(cls):
+        """
+        This method returns a compiled regex to match with a url.
```

Comment:
`Returns a compiled regex that matches any URL this job store support imports and exports of.`